### PR TITLE
chore: 🤖 Migrate `form.inut` to `Hds::TextInput::Field`

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12477

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12477)

## Description
- Migrate `form/create-worker-led` form to use `Hds:Text::InputField` instead of `form.input`
- Fix corresponding test

### Before:
image

### After: 
image

## How to Test
- Run this PR locally using a Boundary dev instance, start the admin ui using `ENABLE_MIRAGE=false API_HOST=http://localhost:9200 yarn start`, and authenticate
- Click on "Workers in sidebar navigation" and select the "New" button
- Click on "Register Worker" to ensure that error messaging fails the submit for the `worker _generated_auth_token`
- Follow these instructions from the test plan for [BYOW](https://docs.google.com/spreadsheets/d/1IOwd54cyUqp1G51amSi_5EPwsxsaprXKTqiH5-z-dc0/edit#gid=109027861) to successfully create a `worker`
- Ensure that after clicking "Register Worker" the button at the bottom of the page converts to "Done"
- Ensure that the worker is listed in the workers list page.

:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:

- [x] I have added before and after screenshots for UI changes
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [x] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

File: ui/admin/app/styles/app.scss [hunk 1]
```diff
@@ -299,6 +299,10 @@
     align-items: flex-end;
     display: flex;
     gap: sizing.rems(m);
+
+    [class*='hds-form-field--layout'] {
```
Comment (calcaide) [inline ui/admin/app/styles/app.scss:303]: What is the purpose of overriding HDS styles here?

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs [hunk 2]
```diff
@@ -12,43 +12,61 @@
   <h3>{{t 'resources.worker.form.steps.1.title'}}</h3>
   <div class='worker-create-section'>
     {{#if (feature-flag 'byow-pki-hcp-cluster-id')}}
-      <form.input
-        @name='cluster_id'
+      <Hds::Form::TextInput::Field
```
Comment (calcaide) [inline ui/admin/app/components/form/worker/create-worker-led/index.hbs:15]: Aren't we missing attributes here such as `isInvalid` or `isOptional`? if you think we do not, can you please elaborate?

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs [hunk 3]
```diff
@@ -12,43 +12,61 @@
...
+          <Hds::Link::Inline @href={{doc-url 'worker.manage-workers'}}>
+            {{t 'actions.learn-more'}}
+          </Hds::Link::Inline>
+        </F.HelperText>
+      </Hds::Form::TextInput::Field>
     {{/if}}
 
-    <form.input
-      @name='ip_address'
+    <Hds::Form::TextInput::Field
```
Comment (calcaide) [inline ui/admin/app/components/form/worker/create-worker-led/index.hbs:31]: Same question: https://github.com/hashicorp/boundary-ui/pull/2151/files#r1503424846

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs [hunk 4]
```diff
@@ -12,43 +12,61 @@
...
+      <F.Label>{{t 'resources.worker.form.ip_address.label'}}</F.Label>
+      <F.HelperText>
+        {{t 'resources.worker.form.ip_address.help'}}
+        <Hds::Link::Inline @href={{doc-url 'worker'}}>
+          {{t 'actions.learn-more'}}
+        </Hds::Link::Inline>
+      </F.HelperText>
+    </Hds::Form::TextInput::Field>
+
+    <Hds::Form::TextInput::Field
```
Comment (calcaide) [inline ui/admin/app/components/form/worker/create-worker-led/index.hbs:46]: Same question: https://github.com/hashicorp/boundary-ui/pull/2151/files#r1503424846

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs [hunk 5]
```diff
@@ -12,43 +12,61 @@
...
+      <F.Label>{{t 'resources.worker.form.config_file_path.label'}}</F.Label>
+      <F.HelperText>
+        {{t 'resources.worker.form.config_file_path.help'}}
+      </F.HelperText>
+    </Hds::Form::TextInput::Field>
 
     {{#if (not (feature-flag 'byow-pki-hcp-cluster-id'))}}
-      <form.input
-        @name='initial_upstreams'
+      <Hds::Form::TextInput::Field
```
Comment (calcaide) [inline ui/admin/app/components/form/worker/create-worker-led/index.hbs:59]: Same question: https://github.com/hashicorp/boundary-ui/pull/2151/files#r1503424846

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs [hunk 6]
```diff
@@ -94,15 +112,19 @@
         </F.Control>
         {{#if this.enableRecordingStoragePath}}
           <F.Control>
-            <form.input
-              @name='recording_storage_path'
+            <Hds::Form::TextInput::Field
```
Comment (calcaide) [inline ui/admin/app/components/form/worker/create-worker-led/index.hbs:115]: Same question: https://github.com/hashicorp/boundary-ui/pull/2151/files#r1503424846

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs [hunk 7]
```diff
@@ -160,26 +182,27 @@
...
+      </Hds::Form::TextInput::Field>
       {{#if (can 'save worker' @model)}}
-        <Rose::Button
-          @onSubmit={{true}}
-          @style='primary'
+        <Hds::Button
+          @text={{t 'resources.worker.actions.register'}}
+          type='submit'
+          disabled={{@model.cannotSave}}
           class='nowrap'
```
Comment (calcaide) [inline ui/admin/app/components/form/worker/create-worker-led/index.hbs:204]: Why do we need this class for an HDS button?

**Review by calcaide — CHANGES_REQUESTED** [review]
Left comments and the current implementation broke the "Worker Auth Registration Request".

Screenshot of the broken implementation:
Screenshot from 2024-02-26 15-16-14

Screenshot with the old implementation working:
Screenshot from 2024-02-26 15-18-09


Can you provide a test procedure to fully test this PR does not brake the worker creation form?

File: ui/admin/app/styles/app.scss (hunk 1 above)
Comment (dvonanderson) [reply ui/admin/app/styles/app.scss:303]: @calcaide the `Hds` wrapper for text input field has some predefined margin on the bottom that is causing the button and the input field to not line up correctly, this selector removes that margin.
Screenshot 2024-02-27 at 09 23 29

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs (hunk 2 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:15]: @calcaide the only field that returns an error on creating a `-worker-generated-auth-token`, all other fields are output only.  However they are optional, in which I fixed in latest commit.  The error messaging for `-worker-generated-auth-token` is also fixed in next commit.

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs (hunk 3 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:31]: @calcaide https://github.com/hashicorp/boundary-ui/pull/2151#discussion_r1504472569

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs (hunk 4 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:46]: @calcaide https://github.com/hashicorp/boundary-ui/pull/2151#discussion_r1504472569

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs (hunk 5 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:65]: @calcaide https://github.com/hashicorp/boundary-ui/pull/2151#discussion_r1504472569

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs (hunk 6 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:127]: @calcaide https://github.com/hashicorp/boundary-ui/pull/2151#discussion_r1504472569

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs [hunk 8]
```diff
@@ -160,26 +182,27 @@
   <h3>{{t 'resources.worker.form.steps.3.title'}}</h3>
   <div class='worker-create-section'>
     <div class='worker-auth-token'>
-      <form.input
-        @name='worker_auth_registration_request'
+      <Hds::Form::TextInput::Field
```
[the remark this answers is not in the document]
Comment (dvonanderson) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:197]: @calcaide https://github.com/hashicorp/boundary-ui/pull/2151#discussion_r1504472569

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs (hunk 3 above)
Comment (calcaide) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:33]: I see you add `@isOptional`

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs (hunk 4 above)
Comment (calcaide) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:50]: I see you add `@isOptional`

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs (hunk 5 above)
Comment (calcaide) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:65]: I see you add `@isOptional`

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs (hunk 8 above)
[the remark this answers is not in the document]
Comment (calcaide) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:197]: I see you add `@isOptional`

File: ui/admin/app/components/form/worker/create-worker-led/index.hbs (hunk 7 above)
Comment (calcaide) [reply ui/admin/app/components/form/worker/create-worker-led/index.hbs:227]: Gotcha, in this use case makes sense.

**Review by calcaide — APPROVED** [review]
LGTM!! Good migration :raised_hands:

**Review by cameronperera — APPROVED** [review]

**Pull request merged**